### PR TITLE
Remove explicit `constraint_constants` from `Pickles.compile`

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR removes the Mina-specific `constraint_constants` from `Pickles.compile`, allowing other uses (e.g. o1js) to omit these irrelevant values.

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

**Comment by mrmr1993** [comment]
!ci-build-me

**Review by mitschabaude — APPROVED** [review]
love how it cleans up the pickles tests!

File: src/lib/pickles/compile.ml [hunk 1]
```diff
@@ -381,18 +382,40 @@ struct
          * _ =
    fun ~self ~cache ~proof_cache ?disk_keys ?override_wrap_domain
        ?override_wrap_main ?(num_chunks = 1) ~branches:(module Branches)
-       ~max_proofs_verified ~name ~constraint_constants ~public_input
+       ~max_proofs_verified ~name ?constraint_constants ?commits ~public_input
        ~auxiliary_typ ~choices () ->
     let snark_keys_header kind constraint_system_hash =
+      let constraint_constants : Snark_keys_header.Constraint_constants.t =
+        match constraint_constants with
```
Comment (emberian) [inline src/lib/pickles/compile.ml:389]: I hate that this looks nicer than `Option.value`

**Review by emberian — APPROVED** [review]

**Comment by emberian** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Pull request merged**